### PR TITLE
新增了一个 使用 protosubff的demo 并解决了心跳问题

### PR DESCRIPTION
**Pull request by chenyu2016**
新增了一个 使用 protosubff的demo 
并发现了一个心跳出错并导致断开的bug
做个一个临时的修复
在 org.dubbo.samples.protosubff.patch.ProtosubffPatch
描述了产生问题的原因 和临时解决方案

### CODE REVIEW

**Comment by chenyu2016** [comment]
DOCKER> [dubbo-samples-generic-impl-provider:2.7.3-SNAPSHOT]: Start container d44d09f5043a
[ERROR] DOCKER> [dubbo-samples-generic-impl-provider:2.7.3-SNAPSHOT]: Timeout after 10045 ms while waiting on log out 'dubbo service started
这个不知道如何解决 CI 没有 pass

File: dubbo-samples-protosubff/dubbo-samples-protosubff-api/src/main/java/org/dubbo/samples/protosubff/patch/ProtosubffPatch.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ * dubbo 2.7.1
+ * protocol 使用 dubbo 序列化设置为 protosubff
+ * 别的为默认
+ * 
+ * 
+ * @author: chenyu
+ * @date:   2019年7月16日 上午10:23:58 
+ *
+ */
+public class ProtosubffPatch {
```
Comment (beiwei30) [inline dubbo-samples-protosubff/dubbo-samples-protosubff-api/src/main/java/org/dubbo/samples/protosubff/patch/ProtosubffPatch.java]: would you mind to submit pull request for the issue you found to dubbo: https://github.com/apache/dubbo/pulls

File: dubbo-samples-protosubff/dubbo-samples-protosubff-provider/src/main/java/org/dubbo/samples/protosubff/provider/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.dubbo.samples.protosubff.provider.service;
+
+import java.util.UUID;
+
+import org.apache.dubbo.config.annotation.Service;
+import org.dubbo.samples.protosubff.api.IUserService;
+import org.dubbo.samples.protosubff.domain.UserInfo;
+
+/**
```
Comment (beiwei30) [inline dubbo-samples-protosubff/dubbo-samples-protosubff-provider/src/main/java/org/dubbo/samples/protosubff/provider/service/UserService.java]: * use English to comment
* add apache license header
* remove author info

Comment (chenyu2016) [reply dubbo-samples-protosubff/dubbo-samples-protosubff-provider/src/main/java/org/dubbo/samples/protosubff/provider/service/UserService.java]: Thank you very much for the fact that I have complied with the request and submitted it.

**Comment by beiwei30** [comment]
https://github.com/apache/dubbo/pull/4645

**Comment by beiwei30** [comment]
@chenyu2016 Pls. check CI faiulre:

```
[ERROR] Failures: 
[ERROR]   ExchangeCodecTest.test_Decode_Error_Request_Object:296 expected: <true> but was: <false>
[ERROR]   ExchangeCodecTest.test_Decode_Return_Request_Event_Object:235 expected: <org.apache.dubbo.remoting.codec.TelnetCodecTest$Person@3c1> but was: <null>
[ERROR]   ExchangeCodecTest.test_Decode_Return_Request_Event_String:250 expected: <R> but was: <null>
[ERROR]   ExchangeCodecTest.test_Decode_Return_Request_Object:278 expected: <org.apache.dubbo.remoting.codec.TelnetCodecTest$Person@3c1> but was: <null>
```

**Comment by beiwei30** [comment]
will merge and modify.

**Pull request merged**